### ISSUE TITLE
num_classes mismatch: pretrain weights has 90 classes, but your model has 93 classes reinitializing detection head with 90 classes

### ISSUE THREAD

**Issue by shelfex-abhishek** [roboflow/rf-detr#370]
### Search before asking

- [x] I have searched the RF-DETR issues and found no similar bug report.

### Bug

when i try to load my custom model which is trained on the  around 93 classes but after loading its initialize the  head to the 90 classes in future i am supposes to train the  rfdetrlarge model in 200 classes
so ho can i tackle this issue  please i need the solution asap.

### Environment

rfdetr latest

### Minimal Reproducible Example

'''''

### Are you willing to submit a PR?

- [ ] Yes, I'd like to help by submitting a PR!

**Comment by ChisenZhang** [comment]
how to fix this bug?

**Comment by SlippyDong** [comment]
Not sure if it's useful, but I found some discrepancies in the SDK ONNX medium model as well:

When I download and run inference with the rf-detr-medium onnx file that's downloaded and cached by the SDK, it comes with class_names.txt and environment.json.

When I check the model's input shape, it shows: Input shape: 1x3x512x512 | static, while the json has:
`"PREPROCESSING": "{\"resize\": {\"enabled\": true, \"format\": \"Stretch to\", \"height\": \"576\", \"width\": \"576\"}}",`
No problem, the model's shape overrides the json so that doesn't matter much.

Then looking at the class_names.txt and CLASS_MAP in the environments.json, I notice that they don't match and on top of that the text file contains numerous duplicate classes, counting up to a total of 91, while the json CLASS_MAP seems to count from 1-90, with gaps in the numbering here and there which seem to match the rows with duplicate classes in the text file.

Upon running inference tests, this caused mislabeling where "person" would be labeled as "bicycle" etc. I can apply a shift-by-one function to handle this discrepancy, but it's just confusing as I expected perhaps the "coco" row to be a "no-object" for background handling.

as for you @ChisenZhang, perhaps you can check the .pt classification head. logits.shape[-1] shows how many classes the model predicts (includes the “no object” class). You can inspect logits.softmax(-1) to see which indices fire.

[environment.json](https://github.com/user-attachments/files/22725304/environment.json)
[class_names.txt](https://github.com/user-attachments/files/22725303/class_names.txt)

**Comment by 576** [comment]
这是来自QQ邮箱自动回复邮件.
您的邮件我已经收到，谢谢祝您愉快！！！

**Comment by 576** [comment]
这是来自QQ邮箱自动回复邮件.
您的邮件我已经收到，谢谢祝您愉快！！！
